src/extensions/savetoLaserfiche/GetDocumentDataDialog.tsx

```diff
@@ -0,0 +1,404 @@
+} from '../../webparts/constants';
+import { ISPDocumentData } from '../../Utils/Types';
+import { Navigation } from 'spfx-navigation';
+import { SPHttpClient, SPHttpClientResponse } from '@microsoft/sp-http';
+import {
+  ActionTypes,
+  ProfileConfiguration,
+  SPProfileConfigurationData,
+} from '../../webparts/laserficheAdminConfiguration/components/ProfileConfigurationComponents';
+import {
+  IPostEntryWithEdocMetadataRequest,
+  FieldToUpdate,
+  IPutFieldValsRequest,
+  PutFieldValsRequest,
+  IValueToUpdate,
+  TemplateFieldInfo,
+  ValueToUpdate,
+  WFieldType,
+} from '@laserfiche/lf-repository-api-client';
+import { IListItem } from '../../webparts/laserficheAdminConfiguration/components/IListItem';
+import { getSPListURL } from '../../Utils/Funcs';
+import SaveToLaserficheCustomDialog from './SaveToLaserficheDialog';
+import { BaseComponentContext } from '@microsoft/sp-component-base';
+import LoadingDialog from './CommonDialogs';
+
+interface ProfileMappingConfiguration {
+  id: string;
+  SharePointContentType: string;
+  LaserficheContentType: string;
+  toggle: boolean;
+}
+
+const signInPageRoute = '/SitePages/LaserficheSpSignIn.aspx';
+
+export class GetDocumentDataCustomDialog extends BaseDialog {
+  successful = false;
+
+  constructor(
+    private fileInfo: {
+      fileName: string;
+      spContentType: string;
+      spFileUrl: string;
+      fileId: string;
+    },
+    private context: BaseComponentContext
+  ) {
+    super();
+  }
+
+  showNextDialog = (data: ISPDocumentData) => {
+    const saveToLfDialog = new SaveToLaserficheCustomDialog(data, () => this.close());
+    this.secondaryDialogProvider.show(saveToLfDialog).then(() => {
+      if (!saveToLfDialog.successful) {
+        Navigation.navigate(
+          this.context.pageContext.web.absoluteUrl + signInPageRoute,
+          true
+        );
+      }
+    });
+  };
+
+  public render(): void {
+    const element: React.ReactElement = (
+      <GetDocumentDialogData
+        spFileInfo={this.fileInfo}
+        context={this.context}
+        showSaveToDialog={this.showNextDialog}
+      />
+    );
+    ReactDOM.render(element, this.domElement);
+  }
+
+  protected onAfterClose(): void {
+    ReactDOM.unmountComponentAtNode(this.domElement);
+    super.onAfterClose();
+  }
+}
+
+const FOLLOWING_SP_FIELDS_BLANK_MAPPED_TO_REQUIRED_LF_FIELDS =
+  'The following SharePoint field values are blank and are mapped to required Laserfiche fields:';
+const PLEASE_FILL_OUT_REQUIRED_FIELDS_TRY_AGAIN =
+  'Please fill out these required fields and try again.';
+
+function GetDocumentDialogData(props: {
+  showSaveToDialog: (fileData: ISPDocumentData) => void;
+  spFileInfo: {
+    fileName: string;
+    spContentType: string;
+    spFileUrl: string;
+    fileId: string;
+  };
+  context: BaseComponentContext;
+}) {
+  const [missingFields, setMissingFields] = React.useState<
+    undefined | SPProfileConfigurationData[]
+  >(undefined);
+
+  if (missingFields) {
```

Comment:
anytime setMissingFields is called/updated the component will re-render so this will run again.
This updates what is passed into the missing Fields dialog, could probably be done a level down